addons/website/tests/test_lang_url.py

```diff
@@ -106,6 +106,11 @@ def test_06_reroute_unicode(self):
         res = self.url_open('/fr/path?привет=1')
         self.assertEqual(res.status_code, 404, "Rerouting didn't crash because of unicode query-string")
 
+    def test_07_nolang_prefix_underscore(self):
+        res = self.url_open('/_not_a_lang', allow_redirects=False)
+        self.assertEqual(res.status_code, 404, "Should not consider /_not_a_lang as a lang")
+        self.assertTrue(res.url.endswith('/_not_a_lang'), "Should use /_not_a_lang as the path and not a lang")
```

Comment:
I've added a tool to make URL comparison in tests easier. It takes care of adding the scheme and  hostname in case they are missing, hope you like it :D
```suggestion
        self.assertURLEqual(res.url, '/_not_a_lang', "Should use /_not_a_lang as the path and not a lang")
```